backbone/src/main/java/org/researchstack/backbone/DataProvider.java

```diff
@@ -143,13 +160,24 @@ public static void init(DataProvider instance)
 
     /**
      * This method is responsible in uploading the user consent information (e.g. Name, Birthdate,
-     * Signature) to the backend
+     * Signature) to the backend.  Usually, this is done by looking into the TaskResult
+     * object and filling up the ConsentSignature and then calling the method below this
+     * with the signature parameter
      *
      * @param context android context
      */
     public abstract void uploadConsent(Context context, TaskResult consentResult);
```

Comment:
My initial thought was to deprecate this method, since it encourages coupling the DataProvider with the Results of the StepLayouts. What do you guys think?